src/MediaBundleForm.php

```diff
@@ -209,6 +215,7 @@ protected function copyFormValuesToEntity(EntityInterface $entity, array $form,
   public function save(array $form, FormStateInterface $form_state) {
     /** @var  \Drupal\media_entity\MediaBundleInterface $bundle */
     $bundle = $this->entity;
+    $bundle->setNewRevision($form_state->getValue('new_revision'));
```

Comment:
This line shouldn't be needed. `copyFormValuesToEntity` should handled that in `parent::copyFormValuesToEntity($entity, $form, $form_state);`